Assets/Scripts/Level generation scripts/TriggerInteractableScript.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TriggerInteractableScript : MonoBehaviour
+{
+    [SerializeField] private Animator _animator;
+    [SerializeField] private float _mapTime = 2;
+    private static readonly int Activated = Animator.StringToHash("Activated");
+    [SerializeField] private float cooldown;
+    private float _timer = 0;
+    
+    private bool _triggered = false;
+    private bool _mapTriggered = false;
+
+    private void OnTriggerExit(Collider other)
+    {
+        if (other.gameObject.name.ToLower().Contains("player") && _triggered)
+        {
+            _animator.SetBool(Activated, true);
+            _timer = 0.0f;
+            StartCoroutine(RefreshCluster(_mapTime));
+        }
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        if (other.gameObject.name.ToLower().Contains("player") && Input.GetButtonDown("Jump"))
+        {
+            _triggered = true;
+        }
+    }
```

Comment:
Is this even necessary if you have the same functionality in `OnTriggerStay` ?

Reply:
It's more of a safety net honestly. It makes the window to hit the button a bit larger. it's not necessary per say, but i put it for more consistency

Reply:
Sounds good!